### PR TITLE
Finale 2.1 revamp

### PR DESCRIPTION
**Pull request by AngrySoundTech**
Opened on behalf of @jaymsDooku

### CODE REVIEW

File: paper/src/main/java/com/github/maxopoly/finale/FinaleManager.java [hunk 1]
```diff
@@ -4,11 +4,18 @@
 import com.github.maxopoly.finale.combat.AsyncPacketHandler;
 import com.github.maxopoly.finale.combat.CPSHandler;
 import com.github.maxopoly.finale.combat.CombatConfig;
-import com.github.maxopoly.finale.misc.ArmourModifier;
-import com.github.maxopoly.finale.misc.SaturationHealthRegenHandler;
-import com.github.maxopoly.finale.misc.WeaponModifier;
+import com.github.maxopoly.finale.combat.SprintHandler;
+import com.github.maxopoly.finale.misc.*;
```
Comment (Diet-Cola) [inline paper/src/main/java/com/github/maxopoly/finale/FinaleManager.java:8]: \* imports bad

File: paper/src/main/java/com/github/maxopoly/finale/combat/SprintHandler.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package com.github.maxopoly.finale.combat;
+
+import com.github.maxopoly.finale.Finale;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerQuitEvent;
+
+import java.util.*;
```
Comment (Diet-Cola) [inline paper/src/main/java/com/github/maxopoly/finale/combat/SprintHandler.java:10]: as above

File: paper/src/main/java/com/github/maxopoly/finale/command/AllyCommand.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
+package com.github.maxopoly.finale.command;
+
+import co.aikar.commands.BaseCommand;
+import co.aikar.commands.annotation.*;
```
Comment (Diet-Cola) [inline paper/src/main/java/com/github/maxopoly/finale/command/AllyCommand.java:4]: \*

File: paper/src/main/java/com/github/maxopoly/finale/listeners/BlockListener.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+	@EventHandler(priority = EventPriority.HIGH)
+	public void onPlace(BlockPlaceEvent event) {
+		Player player = event.getPlayer();
+		CombatTagPlusManager ctpManager = Finale.getPlugin().getCombatTagPlusManager();
+		if (!ctpManager.isTagged(player)) {
+			return;
+		}
+
+		System.out.println("COMBAT TAGGED PLAYER");
```
Comment (Diet-Cola) [inline paper/src/main/java/com/github/maxopoly/finale/listeners/BlockListener.java:35]: sout

File: paper/src/main/java/com/github/maxopoly/finale/listeners/ShieldListener.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+	@EventHandler
+	public void onShieldBash(PlayerInteractEvent event) {
+		if (event.getAction() == Action.RIGHT_CLICK_AIR || event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+			//System.out.println("RIGHT CLICKING");
+			ItemStack item = event.getItem();
+			if (item == null || item.getType() != Material.SHIELD) {
+				return;
+			}
+			//System.out.println("RIGHT CLICKING ON SHIELD");
```
Comment (Diet-Cola) [inline paper/src/main/java/com/github/maxopoly/finale/listeners/ShieldListener.java:37]: sout comments

File: paper/src/main/java/com/github/maxopoly/finale/listeners/TridentListener.java [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+				return;
+			}
+
+			if (trident.getTicksLived() < 60) {
+				return;
+			}
+
+			if (returnTrident.getUniqueId().equals(trident.getUniqueId())) {
+				ItemStack offhand = shooter.getInventory().getItemInOffHand();
+				//System.out.println(offhand);
```
Comment (Diet-Cola) [inline paper/src/main/java/com/github/maxopoly/finale/listeners/TridentListener.java:148]: as above

File: paper/src/main/java/com/github/maxopoly/finale/listeners/WarpFruitListener.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+			if (warpFruitTracker.timewarp(player)) {
+				if (warpFruitTracker.isSpectralWhileChanneling() && player.hasPotionEffect(PotionEffectType.GLOWING)) {
+					player.removePotionEffect(PotionEffectType.GLOWING);
+				}
+				event.setCancelled(true);
+			}
+		}
+	}
+
+//	@EventHandler
```
Comment (Diet-Cola) [inline paper/src/main/java/com/github/maxopoly/finale/listeners/WarpFruitListener.java:107]: is this needed?

File: paper/src/main/java/com/github/maxopoly/finale/misc/ally/AllyHandler.java [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+
+				if (loc == null) {
+					loc = sourcePlayer.getEyeLocation().clone().add(dir);
+
+					loc.getWorld().spawnParticle(Particle.REDSTONE, loc, 1, dustOptions);
+				} else {
+					Location prevLoc = loc;
+					Location newLoc = loc.clone().add(dir);
+
+					/*double particles = 10;
```
Comment (Diet-Cola) [inline paper/src/main/java/com/github/maxopoly/finale/misc/ally/AllyHandler.java:186]: more comments

**Comment by okx-code** [comment]
Please migrate to monorepo https://github.com/CivMC/Civ

**Pull request closed without merging**